src/openforms/variables/models.py

```diff
@@ -0,0 +1,101 @@
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+import jq
+from rest_framework.serializers import ValidationError as DRFValidationError
+
+from openforms.forms.api.validators import JsonLogicValidator
+
+from .constants import DataMappingTypes, ServiceFetchMethods
+from .validators import HeaderValidator
+
+
+class ServiceFetchConfiguration(models.Model):
+    service = models.ForeignKey(
+        "zgw_consumers.Service",
+        on_delete=models.PROTECT,
+    )
+    path = models.CharField(
+        _("path"),
+        max_length=250,
+        blank=True,  # allow "" for the root, but not None
+        default="",
+        help_text=_("path relative to the Service API root"),
+    )
+    method = models.CharField(
+        _("HTTP method"),
+        max_length=4,
+        choices=ServiceFetchMethods.choices,
+        default=ServiceFetchMethods.get,
+        help_text=_("POST is allowed, but should not be used to mutate data"),
+    )
+    headers = models.JSONField(
+        _("HTTP request headers"),
+        blank=True,
+        null=True,
+        help_text=_(
+            "Additions and overrides for the HTTP request headers as defined in the Service."
+        ),
+        validators=[HeaderValidator()],
+    )
+    query_params = models.TextField(
+        _("HTTP query string"),
+        blank=True,
+        default="",
+    )
+    body = models.JSONField(
+        _("HTTP request body"),
+        blank=True,
+        null=True,
+        help_text=_(
+            'Request body for POST requests (only "application/json" is supported)'
+        ),
+    )
+    data_mapping_type = models.CharField(
+        _("mapping expression language"),
+        max_length=10,
+        blank=True,
+        null=True,
```

Comment:
don't use `null=True` for text-based fields, as comparing/querying for empty values has some gotchas. `blank=True` is sufficient.